test/integration/features/addon-registry-route.feature

```diff
@@ -0,0 +1,45 @@
+@addon-registry-route @addon
+Feature: registry-route add-on
+registry-route exposes private docker registry,
+so that user can push their images to private registry and delpoy it.
+
+  Scenario: As a user, I can enable the registry-route add-on
+    Given user starts shell instance on host machine
+     When executing "minishift addons enable registry-route" succeeds
+     Then stdout should contain "Add-on 'registry-route' enabled"
+
+  Scenario: As a user, I can starts Minishift
+    Given Minishift has state "Does Not Exist"
+      And image caching is disabled
+     When executing "minishift start --iso-url centos" succeeds
+     Then Minishift should have state "Running"
+      And stdout should contain
+      """
+      Add-on 'registry-route' created docker-registry route.
+      """
+
+  Scenario: As a user, I can pull app from private OpenShift docker registry
```

Comment:
This scenario is too lengthy.

```
  Scenario: As a user, I can able to see the registry container running
    Given Minishift has state "Running"
      And executing "minishift docker-env" in host shell succeeds
      And evaluating stdout of the previous command in host shell
      And executing "minishift oc-env" in host shell succeeds
      And evaluating stdout of the previous command in host shell
     Then with up to "5" retries with wait period of "6" seconds container image "origin-docker-registry" should be "running"

  Scenario: As a user, I can able to login to OpenShift integrated docker registry
    Given Minishift has state "Running"
     When executing "docker login -u developer -p `oc whoami -t` docker-registry-default.$(minishift ip).nip.io" in host shell succeeds
     Then stdout of host shell should contain "Login Succeeded"
    
  Scenario: As a user, I can pull application image from docker hub and push to OpenShift integrated docker registry
   Given Minishift has state "Running"
     And executing "docker pull bitnami/nginx" in host shell succeeds
     And executing "docker tag bitnami/nginx docker-registry-default.$(minishift ip).nip.io/myproject/nginx" in host shell succeeds
    When executing "docker push docker-registry-default.$(minishift ip).nip.io/myproject/nginx" in host shell succeeds
     And executing "oc get is" in host shell succeeds
    Then stdout of host shell contains "nginx"
 
  Scenario: As a user, I can deploy application using previously pushed application image
   Given executing "oc new-app --image-stream=nginx --name=nginx" in host shell succeeds
     And stdout of host shell contains "Success"
    When executing "oc expose svc nginx" in host shell succeeds
     And executing "oc set probe dc/nginx --readiness --get-url=http://:8080" in host shell succeeds
     And service "nginx" rollout successfully within "1200" seconds
    Then with up to "10" retries with wait period of "500ms" the "status code" of HTTP request to "/" of service "nginx" in namespace "myproject" is equal to "200"
    Then with up to "10" retries with wait period of "500ms" the "body" of HTTP request to "/" of service "nginx" in namespace "myproject" contains "Welcome to nginx!"
```

Reply:
Yes we can split the scenario, however this is more granular IMO and somehow the feature meaning is missing. 

For example ```Scenario: As a user, I can able to see the registry container running``` is not scenario for registry route, this is moreover a steps in the scenario that confirms that before docker login docker-registry container should up and running 

So we can split this big scenario into 2 scenarios. For example 
```
Scenario: As a user, I can login to OpenShift integrated docker registry
Scenario: As a user, I can push pulled application image to OpenShift integrated docker registry and deploy it
```